src/pages/index.js

```diff
@@ -4,128 +4,187 @@ import FormValidator from "../components/FormValidator.js";
+    "Content-Type": "application/json",
+  },
+});
 const editProfileValidator = new FormValidator(config, "#profile-edit-form");
 const addCardValidator = new FormValidator(config, "#add-card-form");
-
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
+const editAvatarValidator = new FormValidator(config, "#edit-avatar-form");
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 
 const addCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleEditAvatarSubmit
+);
 const previewImagePopup = new PopupWithImage("#preview-image-modal");
+const deleteConfirmationPopup = new PopupWithConfirmation(
+  "#delete-confirmation-modal"
+);
 const profileUserInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+/* Event Handlers */
+function handleEditAvatarSubmit(inputValues) {
+  editAvatarPopup.renderLoadingSave(true);
+  api
+    .updateAvatar(inputValues.link)
+    .then((res) => {
+      profileUserInfo.setUserAvatar(res.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}. Failed to edit avatar :(`); // log the error to the console
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoadingSave(false);
+    });
+}
+
 function handleImageClick(data) {
   previewImagePopup.open(data.name, data.link);
 }
 
-/* Event Handlers */
 function handleProfileEditSubmit(inputValues) {
+  editProfilePopup.renderLoadingSave(true);
   //should be handled by userinfo class
-  profileUserInfo.setUserInfo(inputValues.title, inputValues.description);
+
+  api
+    .updateProfileInfo(inputValues) //{title: ddd, description: kk}
+    .then((data) => {
+      profileUserInfo.setUserInfo(inputValues.title, inputValues.description);
+      editProfilePopup.close();
+      //profileUserInfo.setUserInfo(data.name, data.about);
+      //console.log("Name:", data.name, "Job:", data.about);
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}. Failed to update profile info :(`); // log the error to the console
+    })
+    .finally(() => {
+      editProfilePopup.renderLoadingSave(false);
+    });
+}
+
+//likes button
+function handleLikingIcon(card) {
+  //fetch to like card on server
+  api
+    .likeCard(card._id)
+
+    //if successful. we like card locally
+    .then(() => {
+      card.fillInLikeIcon();
+      card.isLiked = false;
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}. Failed to like card :(`);
+    });
+}
+//unlikes button
+function handleUnlikingIcon(card) {
+  //delete like on server
+  api
+    .dislikeCard(card._id)
+    //if successfull, unlike card locally
+    .then(() => {
+      card.clearLikeIcon();
+      this.isLiked = true;
```

Comment:
`this` should **not** be used outside classes in functions in `index.js` . This context is easy to lose. You need to find other ways of passing data.